frontend/src/common/queries/collections.ts

```diff
@@ -407,28 +449,33 @@ export function useEditCollection(collectionID?: Collection["id"]) {
   });
 
   return useMutation(editCollection, {
-    onSuccess: (newCollection) => {
-      queryClient.setQueryData(
-        [USE_COLLECTION, collectionID, VISIBILITY_TYPE.PRIVATE, collections],
-        () => {
-          let revision_diff;
-          if (isTombstonedCollection(newCollection)) {
-            return newCollection;
-          } else if (
-            !isTombstonedCollection(collection) &&
-            !isTombstonedCollection(publishedCollection) &&
-            collection?.has_revision &&
-            publishedCollection
-          ) {
-            revision_diff = checkForRevisionChange(
-              newCollection,
+    onSuccess: ({ collection: newCollection }) => {
+      // Check for updated collection: it's possible server-side validation errors have occurred where the error has
+      // been swallowed (allowing error messages to be displayed on the edit form) and success flow is executed even
+      // though update did not occur.
+      if (newCollection) {
```

Comment:
Wow thanks so much for catching this bug! :pray: CC @seve for knowledge sharing!

Also for readability should we use a guard clause instead? This way we don't indent the whole code block one level deeper and no need to scan for the non-existing `else` block at the bottom :smile: 

```ts
if (!newCollection) return;

queryClient.setQueryData...
```

Reply:
Guard cause is much cleaner; I'll make the update. Thanks Timmy!